src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,144 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    static class Node<T> {
+        private T element;
```

Comment:
Make this class private. And remember that all private classes and methods should go after public.